### PR TITLE
Update dependencies to 0.14.1

### PR DESCRIPTION
**Pull request by tylere**
- Updated versions & removed dependencies to match https://github.com/developmentseed/titiler/blob/0.14.1/src/titiler/core/pyproject.toml
    - async-exit-stack       # removed in 0.4.0
    - async_generator        # removed in 0.4.0
    - dataclasses            # removed in 0.4.0
    - importlib_resources    # removed at 0.11.1

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by tylere** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,24 +17,21 @@ build:
...
-    - importlib_resources >=1.1.0
-    - jinja2 >=2.11.2,<3.0.0
-    - morecantile <3.0.0
+    - fastapi >=0.100.0
+    - geojson-pydantic >=1.0,<2.0
+    - jinja2 >=2.11.2,<4.0.0
     - numpy
-    - pydantic
-    - python >=3.6
+    - pydantic ~=2.0
```
Comment (ocefpaf) [inline recipe/meta.yaml:28]:
```suggestion
    - pydantic >=2.0,<3.dev0
```

**Comment by ocefpaf** [comment]
@conda-forge-admin, please rerender

**Comment by rsignell-usgs** [comment]
Thanks Filipe!

**Comment by ocefpaf** [comment]
> Thanks Filipe!

@tylere did all the work though ;-p

**Pull request merged**